modules/derivation-common/interface.nix

```diff
@@ -103,32 +111,32 @@
       type = t.package;
     };
 
+    depsRequired = l.mkOption {
+      type = t.attrsOf t.bool;
+      default = {};
+    };
+
     /*
-      This allows defining drvs in an encapsulated manner, while maintaining
-        the capability to depend on external attributes
+      Attrset of required dependencies.
+      For example, when `config.depsRequired.bash = true`, then this creates:
+      {
+        options.deps = l.mkOption {
+          description = "spcify a package for the dependency bash"
+          type = t.raw;
+          default = config.depsFrom.bash;
+        }
+      }
     */
-    deps = l.mkOption {
-      description = ''
-        All dependencies of the package. This option should be set by the "outer world" and can be used to inherit attributes from `pkgs` or `inputs` etc.
+    deps =
+      l.mapAttrs
+      mkDepOpt
+      (l.filterAttrs (_: enabled: enabled) config.depsRequired);
```

Comment:
`makeModule` implementation concerns are leaking into the general interface. That's not good.

Shouldn't this be in a `makeModule` module?

Should `deps` be a freeform module?

Reply:
> Shouldn't this be in a makeModule module?

I thought explicitly specifying a list of required deps is something that we want everywhere. `depsRequired` seemed like the simplest way to do that. It's not supposed to be specific to makeModule.

> Should deps be a freeform module?

Will this bring back the capability of specifying `deps` as a function?
I probably need to combine `freeform` and `apply`, I guess?